### PR TITLE
add sd3 draft

### PR DESCRIPTION
**Pull request by sayakpaul**

### CODE REVIEW

File: sd3.md [hunk 1]
```diff
@@ -0,0 +1,248 @@
+---
+title: "Diffusers welcomes Stable Diffusion 3"
+thumbnail: /blog/assets/sd3/thumbnail.png
+authors:
+- user: diffusers
```
Comment (osanseviero) [inline sd3.md:5]: an organization cannot be a user unfortunately

File: sd3.md [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+
+```python
+import torch
+from diffusers import StableDiffusion3Pipeline
+
+pipe = StableDiffusion3Pipeline.from_pretrained("stabilityai/stable-diffusion-3-medium", torch_dtype=torch.float16)
+pipe = pipe.to("cuda")
+
+image = pipe(
+		"A cat holding a sign that says hello world",
```
Comment (osanseviero) [inline sd3.md:73]: some formatting is broken here

File: sd3.md [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+		negative_prompt="",
+    num_inference_steps=28,
+    guidance_scale=7.0,
+).images[0]
+image
+```
+
+![Untitled](%F0%9F%A7%A8%20Diffusers%20welcomes%20Stable%20Diffusion%203%20cb744fcd59bb43e794810ee31f69d54f/Untitled.png)
```
Comment (osanseviero) [inline sd3.md:81]: To switch to documentation-images

**Review by osanseviero — COMMENTED** [review]
Let's go :fire: :fire:

File: sd3.md (hunk 2 above)
Comment (sayakpaul) [reply sd3.md]: Fixed.

File: sd3.md (hunk 1 above)
Comment (sayakpaul) [reply sd3.md:5]: Too bad :(

File: sd3.md (hunk 3 above)
Comment (sayakpaul) [reply sd3.md]: Fixed.

File: sd3.md [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+
+SD3 processes text inputs and pixel latents as a sequence of embeddings. Positional encodings are added to 2x2 patches of the latents which are then flattened into a patch encoding sequence. This sequence, along with the text encoding sequence are fed into the MMDiT blocks, where they are embedded to a common dimensionality, concatenated, and passed through a sequence of modulated attentions and MLPs.
+
+In order to account for the differences between two modalities, the MMDiT blocks use two separate sets of weights to embed the text and image sequences to a common dimensionality.  These sequences are joined before the attention operation, which allows both representations to work in their own space while taking the other one into account during the attention operation [1]. This two-way flow of information between text and image data differs from previous approaches for text-to-image synthesis, where text information is incorporated into the latent via cross-attention with a fixed text representation.
```
Comment (kashif) [inline sd3.md:41]:
```suggestion
In order to account for the differences between the two modalities, the MMDiT blocks use two separate sets of weights to embed the text and image sequences to a common dimensionality.  These sequences are joined before the attention operation, which allows both representations to work in their own space while taking the other one into account during the attention operation [1]. This two-way flow of information between text and image data differs from previous approaches for text-to-image synthesis, where text information is incorporated into the latent via cross-attention with a fixed text representation.       
```

File: sd3.md [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+    _ = pipe(prompt=prompt, generator=torch.manual_seed(1))
+
+# Run Inference
+image = pipe(prompt=prompt, generator=torch.manual_seed(1)).images[0]
+image.save("sd3_hello_world.png")
+```
+
+*Refer [here](https://gist.github.com/sayakpaul/508d89d7aad4f454900813da5d42ca97) for the full script.*
+
+We benchmarked the performance of `torch.compile()`on SD3 on a single 80GB A100 machine  using `fp16` precision and PyTorch 2.3. We ran 10 iterations of a pipeline inference call with 20 tdiffusion steps. We found that the average inference time with the compiled versions of the models was **0.585 seconds,** *a 4X speed up over eager execution*.
```
Comment (kashif) [inline sd3.md:217]:
```suggestion
We benchmarked the performance of `torch.compile()`on SD3 on a single 80GB A100 machine  using `fp16` precision and PyTorch 2.3. We ran 10 iterations of a pipeline inference call with 20 diffusion steps. We found that the average inference time with the compiled versions of the models was **0.585 seconds,** *a 4X speed up over eager execution*.    
```

File: sd3.md [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+
+We benchmarked the performance of `torch.compile()`on SD3 on a single 80GB A100 machine  using `fp16` precision and PyTorch 2.3. We ran 10 iterations of a pipeline inference call with 20 tdiffusion steps. We found that the average inference time with the compiled versions of the models was **0.585 seconds,** *a 4X speed up over eager execution*.    
+
+## Dreambooth and LoRA fine-tuning
+
+Additionally, we’re providing a [DreamBooth](https://dreambooth.github.io/) fine-tuning script for SD3 leveraging [LoRA](https://huggingface.co/blog/lora). The script can be used to efficiently fine-tune SD3 and serves as a reference for implementing rectified flow-based training pipelines. Other popular implementations of rectified flow include [minRF](https://github.com/cloneofsimo/minRF/). 
+
+To get started with the script, first, ensure you have the right setup and a demo dataset available (such as [this one](https://huggingface.co/datasets/diffusers/dog-example)). Refer here (TODO) for details. Install `peft` and `bitsandbytes` and then we’re good to go:
+
+```python
```
Comment (sayakpaul) [inline sd3.md:225]:
```suggestion
```bash
```

File: sd3.md [hunk 7]
```diff
@@ -0,0 +1,253 @@
...
+
+### Training with Rectified Flow Matching
+
+In addition to architectural changes, SD3 applies a [conditional flow-matching objective to train the model](https://arxiv.org/html/2403.03206v1#S2). In this approach, the forward noising process is defined as a [rectified flow](https://arxiv.org/html/2403.03206v1#S3) that connects the data and noise distributions on a straight line. 
+
+The rectified flow-matching sampling process is simpler and performs well when reducing the number of sampling steps. To support inference with SD3, we have introduced a new scheduler (`FlowMatchEulerDiscreteScheduler`) with a rectified flow-matching formulation and Euler method steps. It also implements resolution-dependent shifting of the timestep schedule via a `shift` parameter. Increasing the `shift` value handles noise scaling better for higher resolutions. It is recommended to use `shift=3.0` for the 2B model. 
+
+To quickly try out SD3, refer to the application below: 
+
+<script type="module" src="https://gradio.s3-us-west-2.amazonaws.com/3.45.1/gradio.js"> </script>
```
Comment (osanseviero) [inline sd3.md:53]:
```suggestion
<script type="module" src="https://gradio.s3-us-west-2.amazonaws.com/4.36.1/gradio.js"> </script>
```

File: sd3.md [hunk 8]
```diff
@@ -0,0 +1,253 @@
...
+# Run Inference
+image = pipe(prompt=prompt, generator=torch.manual_seed(1)).images[0]
+image.save("sd3_hello_world.png")
+```
+
+*Refer [here](https://gist.github.com/sayakpaul/508d89d7aad4f454900813da5d42ca97) for the full script.*
+
+We benchmarked the performance of `torch.compile()`on SD3 on a single 80GB A100 machine  using `fp16` precision and PyTorch 2.3. We ran 10 iterations of a pipeline inference call with 20 diffusion steps. We found that the average inference time with the compiled versions of the models was **0.585 seconds,** *a 4X speed up over eager execution*.    
+
+## Dreambooth and LoRA fine-tuning
```
Comment (osanseviero) [inline sd3.md:219]: Nice!

**Review by osanseviero — APPROVED** [review]
Very cool :fire:

**Review by kashif — APPROVED** [review]
